### DISCUSSION TITLE
Live Preview of mini.base16

### DISCUSSION THREAD

**Discussion by mhanberg** [nvim-mini/mini.nvim#1045] [Q&A] [answered]
I am interested in creating a base16 colorscheme that use's my works branding colors as the base. It would be really nice if there was a way to live update a buffer with the changed theme without having to quit and reopen nvim.

Has anyone accomplished this before?

**Comment by echasnovski** [comment]
It's a bit complex, but take a look at the [new dev mode in 'folke/tokyonightnvim](https://github.com/folke/tokyonight.nvim/blob/main/.lazy.lua). It would need adjustments, though, depending on how you organize color scheme files.

Also it utilizes 'mini.hipatterns' from 'mini.nvim', so you should be half way there :)

**Reply by mhanberg** [reply, marked as answer]
Roger, I think I figured something out.

I have a couple autocmds in a lazy plugin (random spot that would get run on bootup) and then another file that has the palette `motchvim.simple.colors`.

We reset the colors module when we reload and simply setup the base16 theme again.

Thanks for the tip!

```lua
local function reload()
	package.loaded["motchvim.simple.colors"] = nil
	require("mini.base16").setup({
		palette = require("motchvim.simple.colors"),
		use_cterm = true,
	})
end
reload = vim.schedule_wrap(reload)

local augroup = vim.api.nvim_create_augroup("colorscheme_dev", { clear = true })
vim.api.nvim_create_autocmd("User", {
	pattern = "VeryLazy",
	group = augroup,
	callback = reload,
})
vim.api.nvim_create_autocmd("BufWritePost", {
	group = augroup,
	pattern = "*simple/colors.lua",
	callback = reload,
})

return {}
```
